caffeinated/modules/events_archive_caff/EED_Events_Archive_Caff.module.php

```diff
@@ -156,6 +156,9 @@ public static function template_settings_form()
     public static function update_template_settings($CFG, $REQ)
     {
         $config = new EE_Events_Archive_Config();
+        $CFG->EED_Events_Archive = $CFG->EED_Events_Archive instanceof EE_Events_Archive_Config
+            ? $CFG->EED_Events_Archive
+            : $config;
```

Comment:
Is it possible that there might be two instances of `EE_Events_Archive_Config` floating about?  Reason I ask is `$config` instantiates a new instance of `EE_Events_Archive_Config`, but that is only set on `$CFG->EED_Events_Archive` if there's already no instance set.

Would it be better to maybe do something like this?

```php
$CFG->EED_Events_Archive = $CFG->EED_Events_Archive instanceof EE_Events_Archive_Config
     ? $CFG->EED_Events_Archive
     : new EE_Events_Archive_Config();
$config = $CFG->EED_Events_Archive;
```

Reply:
> Is it possible that there might be two instances of EE_Events_Archive_Config floating about?

ya maybe. all of the config stuff is horrible old old legacy code so anything is possible.
I'll clean it up a bit more using your idea